spec/std/http/cookie_spec.cr

```diff
@@ -285,9 +279,25 @@ module HTTP
         first.value.should eq("bar")
         second.value.should eq("baz")
       end
+
+      it "parses cookie with spaces in value" do
+        parse_first_cookie(%[key=some value]).value.should eq "some value"
+        parse_first_cookie(%[key="some value"]).value.should eq "some value"
+      end
     end
 
     describe "parse_set_cookie" do
+      it "with space" do
+        cookie = parse_set_cookie("key=value; path=/test")
+        parse_set_cookie("key=value;path=/test").should eq cookie
+        parse_set_cookie("key=value;  \t\npath=/test").should eq cookie
```

Comment:
If I'm not mistaken, there's a test case missing here. A value with leading and trailing whitespace. From [RFC 6265](https://datatracker.ietf.org/doc/html/rfc6265#section-5.2):

> 4.  Remove any leading or trailing WSP characters from the name
       string and the value string.

```crystal
parse_set_cookie("key= \tvalue \t;  \t\npath=/test").should eq cookie
```

Reply:
Changed logic a little based on @straight-shoota's comment. Now only unquoted values are stripped.

Reply:
@anton7c3 what about spaces around the quotes? for example `key=  " value"  `

Reply:
They will be consumed by `\s` around `CookieValue` regex.
The following examples work:
```crystal
        parse_first_cookie(%[key=  " some value  "  ]).value.should eq " some value  "

        cookie = parse_set_cookie(%[key=  " value  "\t ;  \tpath=/test])
        cookie.name.should eq "key"
        cookie.value.should eq " value  "
        cookie.path.should eq "/test"
```

I didn't commit the changes since there are multiple approves already.

Reply:
If you have more useful specs, feel free to amend them as additional commits (no force-push). That's super easy to review and it would include everything in one change (instead of adding them in a follow-up).

Reply:
Added an example mentioned above.